File: spring-boot-java/src/main/java/com/example/springbootjava/document/Movie.java
package com.example.springbootjava.document;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "movies")
@FieldDefaults(level = AccessLevel.PRIVATE)
@Data
public class Movie {

    @Id
    String id;
    String year, plot, poster, title, fullplot;
    List<String> genres, cast;
}

File: spring-webflux-kotlin/src/main/kotlin/com/example/springwebfluxkotlin/document/Movie.kt
package com.example.springwebfluxkotlin.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "movies")
data class Movie(
    var year: String,
    var title: String,
    var genres: List<String?> = emptyList(),
    var cast: List<String?> = emptyList(),
    var plot: String? = null,
    var fullplot: String? = null,
    var poster: String? = null,
    @Id
    var id: String? = null
)
